### PR TITLE
test: add e2e test cases for proxy-rewrite plugin

### LINKED ISSUE #335: add e2e test cases for request-rewrite plugin
See https://github.com/apache/apisix-ingress-controller/issues/321 for more details.

### PR DESCRIPTION
**Pull request by Firstsawyou**
Please answer these questions before submitting a pull request

close #335 

- Why submit this pull request?
- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues

___
### Bugfix
- Description

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

___
### Backport patches
- Why need to backport?

- Source branch

- Related commits and pull requests

- Target branch

### CODE REVIEW

File: test/e2e/plugins/proxy_rewrite.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+		assert.Nil(ginkgo.GinkgoT(), s.CreateResourceFromString(ar))
+
+		err := s.EnsureNumApisixUpstreamsCreated(1)
+		assert.Nil(ginkgo.GinkgoT(), err, "Checking number of upstreams")
+		err = s.EnsureNumApisixRoutesCreated(1)
+		assert.Nil(ginkgo.GinkgoT(), err, "Checking number of routes")
+
+		s.NewAPISIXClient().GET("/hello").WithHeader("Host", "httpbin.org").
+			Expect().
+			Status(200)
```
Comment (tokers) [inline test/e2e/plugins/proxy_rewrite.go:72]: Also check the body, which contains the string "origin".

File: test/e2e/plugins/proxy_rewrite.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+`, backendSvc, backendPorts[0])
+
+		assert.Nil(ginkgo.GinkgoT(), s.CreateResourceFromString(ar))
+
+		err := s.EnsureNumApisixUpstreamsCreated(1)
+		assert.Nil(ginkgo.GinkgoT(), err, "Checking number of upstreams")
+		err = s.EnsureNumApisixRoutesCreated(1)
+		assert.Nil(ginkgo.GinkgoT(), err, "Checking number of routes")
+
+		s.NewAPISIXClient().GET("/hello").WithHeader("Host", "test.com").
```
Comment (tokers) [inline test/e2e/plugins/proxy_rewrite.go:109]: Ditto.

**Comment by tokers** [comment]
Missing cases for regex_uri.

**Comment by tokers** [comment]
Ping @Firstsawyou Any updates?

**Comment by Firstsawyou** [comment]
> Ping @Firstsawyou Any updates?

Being resolved.

**Comment by tokers** [comment]
@Firstsawyou Please fix the e2e failure problem.

File: test/e2e/plugins/proxy_rewrite.go (hunk 1 above)
Comment (Firstsawyou) [reply test/e2e/plugins/proxy_rewrite.go]: updated.

File: test/e2e/plugins/proxy_rewrite.go (hunk 2 above)
Comment (Firstsawyou) [reply test/e2e/plugins/proxy_rewrite.go:111]: updated.

**Comment by Firstsawyou** [comment]
> @Firstsawyou Please fix the e2e failure problem.

Fixed.

File: test/e2e/plugins/proxy_rewrite.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+`, backendSvc, backendPorts[0])
+
+		assert.Nil(ginkgo.GinkgoT(), s.CreateResourceFromString(ar))
+
+		err := s.EnsureNumApisixUpstreamsCreated(1)
+		assert.Nil(ginkgo.GinkgoT(), err, "Checking number of upstreams")
+		err = s.EnsureNumApisixRoutesCreated(1)
+		assert.Nil(ginkgo.GinkgoT(), err, "Checking number of routes")
+
+		s.NewAPISIXClient().GET("/hello/ip").
```
Comment (tokers) [inline test/e2e/plugins/proxy_rewrite.go:153]: Should also add a new test case to verify the proxy-rewrite plugin is not effective, for instance, the regex pattern is not matched.

**Review by tokers — APPROVED** [review]

File: test/e2e/plugins/proxy_rewrite.go (hunk 3 above)
Comment (Firstsawyou) [reply test/e2e/plugins/proxy_rewrite.go:153]: Added.

**Pull request merged**